lib.c

```diff
@@ -323,20 +329,35 @@ int osc_init_sdk(struct osc_env *e, const char *profile, unsigned int flag)
 		osc_str_append_string(&e->endpoint, endpoint);
 	}
 
-	if (!e->ak || !e->sk) {
-		fprintf(stderr, "access key and secret key needed\n");
+	if (auth && (!strcmp(auth, "password") || !strcmp(auth, "basic")))
+		e->auth_method = OSC_PASSWORD_METHOD;
+	else if (auth && strcmp(auth, "accesskey")) {
+		fprintf(stderr, "'%s' invalude auth methode\n", auth);
 		return -1;
 	}
 
-	if (strlen(e->ak) != AK_SIZE || strlen(e->sk) != SK_SIZE) {
-		fprintf(stderr, "Wrong access key or secret key size\n");
-		return -1;
+	if (e->auth_method == OSC_AKSK_METHODE) {
+		if (!e->ak || !e->sk) {
+			fprintf(stderr, "access key and secret key needed\n");
+			return -1;
+		}
+
+		if (strlen(e->ak) != AK_SIZE || strlen(e->sk) != SK_SIZE) {
```

Comment:
strnlen_s ?

Reply:
I'm not sure what it would check, I already check that ak/sk are not null.
if the string lack a 0 termination, then there is already nothing I can do.

So adding those check would only be useful, if there is an non-null-terminated string that go beyond strsz.
but I'm not sure the positive of this would overcome the negative of adding a C11 functions that might be less supported than `strlen`. (C11 glibc support is not the same as C11 compiler support)

Reply:
I've made this issue https://github.com/outscale/osc-sdk-c/issues/23